IllegalArgumentException when unregistering receiver in EditPostActivity

The receiver is only registered in onCreate and unregistered in onDestroy, it's odd that we've gotten into this state to begin with. Will probably just have to surround with a try/catch block to prevent the crash.

```
java.lang.RuntimeException: Unable to destroy activity {org.wordpress.android/org.wordpress.android.ui.posts.EditPostActivity}: java.lang.IllegalArgumentException: Receiver not registered: org.wordpress.android.ui.posts.EditPostActivity$4@413a9f10
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3451)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3469)
       at android.app.ActivityThread.access$1200(ActivityThread.java:141)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1287)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.app.ActivityThread.main(ActivityThread.java:5041)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:511)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.IllegalArgumentException: Receiver not registered: org.wordpress.android.ui.posts.EditPostActivity$4@413a9f10
       at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:657)
       at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1339)
       at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:445)
       at org.wordpress.android.ui.posts.EditPostActivity.onDestroy(EditPostActivity.java:310)
       at android.app.Activity.performDestroy(Activity.java:5273)
       ...
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java
@@ -307,7 +307,13 @@ public class EditPostActivity extends ActionBarActivity implements EditorFragmen
     @Override
     protected void onDestroy() {
         super.onDestroy();
-        unregisterReceiver(mGalleryReceiver);
+
+        try {
+            unregisterReceiver(mGalleryReceiver);
+        } catch (IllegalArgumentException e) {
+            AppLog.d(T.EDITOR, "Illegal state! Can't unregister receiver that was no registered");
+        }
+
         AnalyticsTracker.track(AnalyticsTracker.Stat.EDITOR_CLOSED_POST);
     }
 

--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java
@@ -266,9 +266,6 @@ public class EditPostActivity extends ActionBarActivity implements EditorFragmen
             }
         });
         ActivityId.trackLastActivity(ActivityId.POST_EDITOR);
-
-        registerReceiver(mGalleryReceiver,
-                new IntentFilter(LegacyEditorFragment.ACTION_MEDIA_GALLERY_TOUCHED));
     }
 
     class AutoSaveTask extends TimerTask {
@@ -280,6 +277,10 @@ public class EditPostActivity extends ActionBarActivity implements EditorFragmen
     @Override
     protected void onResume() {
         super.onResume();
+
+        registerReceiver(mGalleryReceiver,
+                new IntentFilter(LegacyEditorFragment.ACTION_MEDIA_GALLERY_TOUCHED));
+
         refreshBlogMedia();
         mAutoSaveTimer = new Timer();
         mAutoSaveTimer.scheduleAtFixedRate(new AutoSaveTask(), AUTOSAVE_INTERVAL_MILLIS, AUTOSAVE_INTERVAL_MILLIS);
@@ -300,6 +301,13 @@ public class EditPostActivity extends ActionBarActivity implements EditorFragmen
     @Override
     protected void onPause() {
         super.onPause();
+
+        try {
+            unregisterReceiver(mGalleryReceiver);
+        } catch (IllegalArgumentException e) {
+            AppLog.d(T.EDITOR, "Illegal state! Can't unregister receiver that was no registered");
+        }
+
         stopMediaUploadService();
         mAutoSaveTimer.cancel();
     }
@@ -307,12 +315,6 @@ public class EditPostActivity extends ActionBarActivity implements EditorFragmen
     @Override
     protected void onDestroy() {
         super.onDestroy();
-
-        try {
-            unregisterReceiver(mGalleryReceiver);
-        } catch (IllegalArgumentException e) {
-            AppLog.d(T.EDITOR, "Illegal state! Can't unregister receiver that was no registered");
-        }
 
         AnalyticsTracker.track(AnalyticsTracker.Stat.EDITOR_CLOSED_POST);
     }

